src/System.Buffers.Experimental/System/Buffers/ReadOnlyBytes.cs

```diff
@@ -10,35 +10,39 @@ namespace System.Buffers
     /// <summary>
     ///  Multi-segment buffer
     /// </summary>
-    public struct ReadOnlyBytes : IReadOnlyBufferList<byte>
+    public readonly struct ReadOnlyBytes : IReadOnlyBufferList<byte>
     {
-        ReadOnlyMemory<byte> _first; // pointer + index:int + length:int
-        IReadOnlyBufferList<byte> _rest;
-        long _totalLength;
+        readonly ReadOnlyMemory<byte> _first; // pointer + index:int + length:int
+        readonly IReadOnlyBufferList<byte> _all;
+        readonly long _totalLength;
 
         static readonly ReadOnlyBytes s_empty = new ReadOnlyBytes(ReadOnlyMemory<byte>.Empty);
 
-        public ReadOnlyBytes(ReadOnlyMemory<byte> first, IReadOnlyBufferList<byte> rest, long length)
+        public ReadOnlyBytes(ReadOnlyMemory<byte> buffer)
         {
-            if(length < 0) throw new ArgumentOutOfRangeException(nameof(length));
-            if (first.Length + (rest == null ? 0 : rest.Length) < length)
-                throw new ArgumentOutOfRangeException(nameof(length), length, string.Format("{0}+{1}", first.Length, rest.Length));
-            _rest = rest;
-            _first = first;
-            _totalLength = length;
+            _first = buffer;
+            _all = null;
+            _totalLength = _first.Length;
         }
 
-        public ReadOnlyBytes(ReadOnlyMemory<byte> first, IReadOnlyBufferList<byte> rest) :
-            this(first, rest, first.Length + (rest == null ? 0 : rest.Length)) { }
+        public ReadOnlyBytes(IReadOnlyBufferList<byte> segments, long length)
```

Comment:
`length` is unused.

Reply:
Ah, thanks; copy and paste is evil :-)

Reply:
nit: probably `ReadOnlyBytes(IReadOnlyBufferList<byte>)` should be placed near `ReadOnlyBytes(IReadOnlyBufferList<byte>, long)` for maintainability.

Reply:
Yeah, will do.